libs/frontend/feature-build/src/lib/dnd/builder-dnd-component.ts

```diff
@@ -86,7 +77,12 @@ const renderImageHover = (
   // Pass the margin of img to the wrapper element, and set the margin of the rendered img
   // to 0 during hover to avoid shifting issue.
   const imgStyleProp = structuredClone(props.style ?? {}) as Record<string, unknown>
-  imgStyleProp['margin'] = '0'
+  // Set the width&height of <img> to 100% because the width&height of the hover wrap (parent) is defined.
+  // In HTML, if the width or height of the parent of <img> is a length value (i.e. 100%, 100px, 1rem, etc.),
+  // the % in the dimensions of <img> will be calculated based on the size of its parent element.
+  imgStyleProp.width = '100%'
+  imgStyleProp.height = '100%'
+  imgStyleProp.margin = '0'
```

Comment:
The image hover issue is taking way longer than I expected 😵. Sorry about that!
The best solution I could find is to get the width&height directly from the bounding rect of the target `<img>`, and use it as the dimensions of the hover wrap. By setting the width&height of the `<img>` to 100%, it'll have exactly the same width&height of its parent element due to how `<img>` works in HTML. I didn't find any information about this on MDN, though. 🤔 

I always thought the width&height (%) of an absolute positioning `<img>` is calculated by the dimensions of its `position: relative` parent, which is actually wrong. No matter if the `<img>` is absolute positioned or not, if the width/height of its direct parent is not a length value, `width: 10%` on a `<img>` will result in `sourceWidth * 10%`. But as long as the width/height of its parent is a length value (i.e. `100px`, `50%`, `0.5rem`, etc), `width: 10%` on a `<img>` will then result in `parentWidthInPx * 10%`.

Reply:
Yeah! I think I have encountered this problem in the past. CSS interactions with auto/initial dimensions can be unexpected, especially when combined with flex containers.

There is still a small jump in height when selecting an image and hovering away, but overall this seems like a big improvement, I think we can move forward!


https://github.com/samatechtw/pubstudio-builder/assets/550982/3ddf88a7-6032-41c8-8095-5ac1d755f986

Reply:
Oh, sorry about that! I think the issue is caused by the `Math.floor()` at `builder-dnd-component:57` and `builder-dnd-component:58` when calculating the width&height to be used on the wrapper div. Removing them seems to make the small jump not that obvious:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/eb4517c2-d4bf-4478-a779-b26b186b64a4

I'll include this in my next PR, or you could do that if that'll make things easier for you, thank you!

Reply:
You can include it, no problem! I am stuck working on my own bug at the moment 😅